File: 백준/Silver/2581. 소수/소수.py
import math
def primenumber(x):
    if x == 1:
        return False
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True
m = int(input())
n = int(input())
arr = list(range(m,n+1))
result = []
for i in arr:
    if primenumber(i) == True:
        result.append(i)
if len(result) == 0:
    print(-1)
else:
    print(sum(result))
    print(min(result))

File: 백준/Bronze/2581. 소수/소수.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;


public class Main {
    public static void main(String[] args) {

        Scanner scan = new Scanner(System.in);
        int start = scan.nextInt();
        int end = scan.nextInt();

        boolean[] isPrime = new boolean[end+1];
        for(int i = 2; i<=end;i++){
            isPrime[i] = true;
        }
        for(int i = 2; i<= Math.sqrt(end); i++){
            if(isPrime[i]){
                for(int j = i*i ; j<=end;j+=i){
                    isPrime[j] = false;
                }
            }
        }
        int sum = 0;
        int min = Integer.MAX_VALUE;
        for(int i = start; i<= end; i++) {
            if (isPrime[i]) {
                sum += i;
                if (i < min) {
                    min = i;
                }
            }
        }
        if (sum == 0){
            System.out.println(-1);
        }else{
            System.out.println(sum);
            System.out.println(min);
        }

    }
}
